{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-15T18:07:35.720860Z",
     "start_time": "2019-03-15T18:07:33.914676Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done!\n"
     ]
    }
   ],
   "source": [
    "%matplotlib inline\n",
    "import matplotlib as mpl\n",
    "from matplotlib import pyplot as plt\n",
    "from matplotlib import gridspec\n",
    "import matplotlib.patheffects as path_effects\n",
    "\n",
    "import StringIO\n",
    "import numpy as np\n",
    "\n",
    "import imp\n",
    "bt = imp.load_source('baltic', '/Users/evogytis/Documents/baltic/baltic.py')\n",
    "\n",
    "import treetime\n",
    "\n",
    "typeface='Helvetica Neue'\n",
    "\n",
    "mpl.rcParams['font.weight']=300\n",
    "mpl.rcParams['axes.labelweight']=300\n",
    "mpl.rcParams['font.family']=typeface\n",
    "mpl.rcParams['font.size']=22\n",
    "\n",
    "masked_set=[]\n",
    "for line in open('/Users/evogytis/Documents/genomic-horizon/data/masked_set.txt','r'):\n",
    "    masked_set.append(line.strip('\\n'))\n",
    "\n",
    "print 'Done!'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use RAxML to infer ML phylogenies for genome and GP."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-13T18:05:57.895176Z",
     "start_time": "2018-04-13T17:46:58.430317Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Makona_600_genome.fasta Makona_600_gp.fasta\n",
      "Makona_600_genome.fasta\n",
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "\n",
      "Just in case you might need it, a mixed model file with \n",
      "model assignments for undetermined columns removed is printed to file Makona_600_genome.part.reduced\n",
      "\n",
      "IMPORTANT WARNING: Alignment validation warnings have been suppressed. Found 16 duplicate sequences and 21 undetermined columns\n",
      "\n",
      "Just in case you might need it, an alignment file with \n",
      "sequence duplicates and undetermined columns removed is printed to file Makona_600_genome.fasta.reduced\n",
      "\n",
      "Alignment has 21 completely undetermined sites that will be automatically removed from the input data\n",
      "\n",
      "\n",
      "This is the RAxML Master Pthread\n",
      "\n",
      "This is RAxML Worker Pthread Number: 1\n",
      "\n",
      "\n",
      "This is RAxML version 8.2.8 released by Alexandros Stamatakis on March 23 2016.\n",
      "\n",
      "With greatly appreciated code contributions by:\n",
      "Andre Aberer      (HITS)\n",
      "Simon Berger      (HITS)\n",
      "Alexey Kozlov     (HITS)\n",
      "Kassian Kobert    (HITS)\n",
      "David Dao         (KIT and HITS)\n",
      "Sarah Lutteropp   (KIT and HITS)\n",
      "Nick Pattengale   (Sandia)\n",
      "Wayne Pfeiffer    (SDSC)\n",
      "Akifumi S. Tanabe (NRIFS)\n",
      "Charlie Taylor    (UF)\n",
      "\n",
      "\n",
      "Alignment has 1595 distinct alignment patterns\n",
      "\n",
      "Proportion of gaps and completely undetermined characters in this alignment: 0.53%\n",
      "\n",
      "RAxML rapid hill-climbing mode\n",
      "\n",
      "Using 4 distinct models/data partitions with joint branch length optimization\n",
      "\n",
      "\n",
      "Executing 1 inferences on the original alignment using 1 distinct randomized MP trees\n",
      "\n",
      "All free model parameters will be estimated by RAxML\n",
      "ML estimate of 25 per site rate categories\n",
      "\n",
      "Likelihood of final tree will be evaluated and optimized under GAMMA\n",
      "\n",
      "GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units\n",
      "\n",
      "Partition: 0\n",
      "Alignment Patterns: 426\n",
      "Name:  cp1\n",
      "DataType: DNA\n",
      "Substitution Matrix: GTR\n",
      "\n",
      "\n",
      "\n",
      "Partition: 1\n",
      "Alignment Patterns: 388\n",
      "Name:  cp2\n",
      "DataType: DNA\n",
      "Substitution Matrix: GTR\n",
      "\n",
      "\n",
      "\n",
      "Partition: 2\n",
      "Alignment Patterns: 402\n",
      "Name:  cp3\n",
      "DataType: DNA\n",
      "Substitution Matrix: GTR\n",
      "\n",
      "\n",
      "\n",
      "Partition: 3\n",
      "Alignment Patterns: 379\n",
      "Name:  ig\n",
      "DataType: DNA\n",
      "Substitution Matrix: GTR\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "RAxML was called as follows:\n",
      "\n",
      "raxml -f d -T 2 -m GTRCAT -q Makona_600_genome.part -c 25 -p 666 -n Makona_600_genome.raxml.tree --silent -s Makona_600_genome.fasta \n",
      "\n",
      "\n",
      "Partition: 0 with name:  cp1\n",
      "Base frequencies: 0.330 0.211 0.200 0.260 \n",
      "\n",
      "Partition: 1 with name:  cp2\n",
      "Base frequencies: 0.308 0.222 0.201 0.268 \n",
      "\n",
      "Partition: 2 with name:  cp3\n",
      "Base frequencies: 0.311 0.224 0.205 0.261 \n",
      "\n",
      "Partition: 3 with name:  ig\n",
      "Base frequencies: 0.328 0.198 0.186 0.287 \n",
      "\n",
      "Inference[0]: Time 610.199599 CAT-based likelihood -35235.772048, best rearrangement setting 20\n",
      "\n",
      "\n",
      "Conducting final model optimizations on all 1 trees under GAMMA-based models ....\n",
      "\n",
      "Inference[0] final GAMMA-based Likelihood: -38761.206491 tree written to file /Users/evogytis/Documents/genomic-horizon/data/alignments/RAxML_result.Makona_600_genome.raxml.tree\n",
      "\n",
      "\n",
      "Starting final GAMMA-based thorough Optimization on tree 0 likelihood -38761.206491 .... \n",
      "\n",
      "Final GAMMA-based Score of best tree -38760.361099\n",
      "\n",
      "Program execution info written to /Users/evogytis/Documents/genomic-horizon/data/alignments/RAxML_info.Makona_600_genome.raxml.tree\n",
      "Best-scoring ML tree written to: /Users/evogytis/Documents/genomic-horizon/data/alignments/RAxML_bestTree.Makona_600_genome.raxml.tree\n",
      "\n",
      "Overall execution time: 1044.032925 secs or 0.290009 hours or 0.012084 days\n",
      "\n",
      "Makona_600_gp.fasta\n",
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "\n",
      "Just in case you might need it, a mixed model file with \n",
      "model assignments for undetermined columns removed is printed to file Makona_600_gp.part.reduced\n",
      "\n",
      "IMPORTANT WARNING: Alignment validation warnings have been suppressed. Found 449 duplicate sequences and 1 undetermined column\n",
      "\n",
      "Just in case you might need it, an alignment file with \n",
      "sequence duplicates and undetermined columns removed is printed to file Makona_600_gp.fasta.reduced\n",
      "\n",
      "Alignment has 1 completely undetermined sites that will be automatically removed from the input data\n",
      "\n",
      "\n",
      "This is the RAxML Master Pthread\n",
      "\n",
      "This is RAxML Worker Pthread Number: 1\n",
      "\n",
      "\n",
      "This is RAxML version 8.2.8 released by Alexandros Stamatakis on March 23 2016.\n",
      "\n",
      "With greatly appreciated code contributions by:\n",
      "Andre Aberer      (HITS)\n",
      "Simon Berger      (HITS)\n",
      "Alexey Kozlov     (HITS)\n",
      "Kassian Kobert    (HITS)\n",
      "David Dao         (KIT and HITS)\n",
      "Sarah Lutteropp   (KIT and HITS)\n",
      "Nick Pattengale   (Sandia)\n",
      "Wayne Pfeiffer    (SDSC)\n",
      "Akifumi S. Tanabe (NRIFS)\n",
      "Charlie Taylor    (UF)\n",
      "\n",
      "\n",
      "Alignment has 184 distinct alignment patterns\n",
      "\n",
      "Proportion of gaps and completely undetermined characters in this alignment: 0.09%\n",
      "\n",
      "RAxML rapid hill-climbing mode\n",
      "\n",
      "Using 3 distinct models/data partitions with joint branch length optimization\n",
      "\n",
      "\n",
      "Executing 1 inferences on the original alignment using 1 distinct randomized MP trees\n",
      "\n",
      "All free model parameters will be estimated by RAxML\n",
      "ML estimate of 25 per site rate categories\n",
      "\n",
      "Likelihood of final tree will be evaluated and optimized under GAMMA\n",
      "\n",
      "GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units\n",
      "\n",
      "Partition: 0\n",
      "Alignment Patterns: 53\n",
      "Name:  cp1\n",
      "DataType: DNA\n",
      "Substitution Matrix: GTR\n",
      "\n",
      "\n",
      "\n",
      "Partition: 1\n",
      "Alignment Patterns: 53\n",
      "Name:  cp2\n",
      "DataType: DNA\n",
      "Substitution Matrix: GTR\n",
      "\n",
      "\n",
      "\n",
      "Partition: 2\n",
      "Alignment Patterns: 78\n",
      "Name:  cp3\n",
      "DataType: DNA\n",
      "Substitution Matrix: GTR\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "RAxML was called as follows:\n",
      "\n",
      "raxml -f d -T 2 -m GTRCAT -q Makona_600_gp.part -c 25 -p 666 -n Makona_600_gp.raxml.tree --silent -s Makona_600_gp.fasta \n",
      "\n",
      "\n",
      "Partition: 0 with name:  cp1\n",
      "Base frequencies: 0.330 0.193 0.308 0.169 \n",
      "\n",
      "Partition: 1 with name:  cp2\n",
      "Base frequencies: 0.295 0.269 0.196 0.239 \n",
      "\n",
      "Partition: 2 with name:  cp3\n",
      "Base frequencies: 0.275 0.273 0.195 0.257 \n",
      "\n",
      "Inference[0]: Time 64.221923 CAT-based likelihood -3491.884510, best rearrangement setting 15\n",
      "\n",
      "\n",
      "Conducting final model optimizations on all 1 trees under GAMMA-based models ....\n",
      "\n",
      "\n",
      "WARNING the alpha parameter with a value of 12.491566 estimated by RAxML for partition number 1 with the name \" cp2\"\n",
      "is larger than 10.000000. You should do a model test and confirm that you actually need to incorporate a model of rate heterogeneity!\n",
      "You can run inferences with a plain substitution model (without rate heterogeneity) by specifyng the CAT model and the \"-V\" option!\n",
      "\n",
      "\n",
      "WARNING the alpha parameter with a value of 28.861716 estimated by RAxML for partition number 1 with the name \" cp2\"\n",
      "is larger than 10.000000. You should do a model test and confirm that you actually need to incorporate a model of rate heterogeneity!\n",
      "You can run inferences with a plain substitution model (without rate heterogeneity) by specifyng the CAT model and the \"-V\" option!\n",
      "\n",
      "\n",
      "WARNING the alpha parameter with a value of 66.647836 estimated by RAxML for partition number 1 with the name \" cp2\"\n",
      "is larger than 10.000000. You should do a model test and confirm that you actually need to incorporate a model of rate heterogeneity!\n",
      "You can run inferences with a plain substitution model (without rate heterogeneity) by specifyng the CAT model and the \"-V\" option!\n",
      "\n",
      "\n",
      "WARNING the alpha parameter with a value of 144.680124 estimated by RAxML for partition number 1 with the name \" cp2\"\n",
      "is larger than 10.000000. You should do a model test and confirm that you actually need to incorporate a model of rate heterogeneity!\n",
      "You can run inferences with a plain substitution model (without rate heterogeneity) by specifyng the CAT model and the \"-V\" option!\n",
      "\n",
      "\n",
      "WARNING the alpha parameter with a value of 324.842266 estimated by RAxML for partition number 1 with the name \" cp2\"\n",
      "is larger than 10.000000. You should do a model test and confirm that you actually need to incorporate a model of rate heterogeneity!\n",
      "You can run inferences with a plain substitution model (without rate heterogeneity) by specifyng the CAT model and the \"-V\" option!\n",
      "\n",
      "\n",
      "WARNING the alpha parameter with a value of 740.749355 estimated by RAxML for partition number 1 with the name \" cp2\"\n",
      "is larger than 10.000000. You should do a model test and confirm that you actually need to incorporate a model of rate heterogeneity!\n",
      "You can run inferences with a plain substitution model (without rate heterogeneity) by specifyng the CAT model and the \"-V\" option!\n",
      "\n",
      "\n",
      "WARNING the alpha parameter with a value of 1000.000000 estimated by RAxML for partition number 1 with the name \" cp2\"\n",
      "is larger than 10.000000. You should do a model test and confirm that you actually need to incorporate a model of rate heterogeneity!\n",
      "You can run inferences with a plain substitution model (without rate heterogeneity) by specifyng the CAT model and the \"-V\" option!\n",
      "\n",
      "\n",
      "WARNING the alpha parameter with a value of 1000.000000 estimated by RAxML for partition number 1 with the name \" cp2\"\n",
      "is larger than 10.000000. You should do a model test and confirm that you actually need to incorporate a model of rate heterogeneity!\n",
      "You can run inferences with a plain substitution model (without rate heterogeneity) by specifyng the CAT model and the \"-V\" option!\n",
      "\n",
      "\n",
      "WARNING the alpha parameter with a value of 1000.000000 estimated by RAxML for partition number 1 with the name \" cp2\"\n",
      "is larger than 10.000000. You should do a model test and confirm that you actually need to incorporate a model of rate heterogeneity!\n",
      "You can run inferences with a plain substitution model (without rate heterogeneity) by specifyng the CAT model and the \"-V\" option!\n",
      "\n",
      "Inference[0] final GAMMA-based Likelihood: -3835.602597 tree written to file /Users/evogytis/Documents/genomic-horizon/data/alignments/RAxML_result.Makona_600_gp.raxml.tree\n",
      "\n",
      "\n",
      "Starting final GAMMA-based thorough Optimization on tree 0 likelihood -3835.602597 .... \n",
      "\n",
      "Final GAMMA-based Score of best tree -3835.602456\n",
      "\n",
      "Program execution info written to /Users/evogytis/Documents/genomic-horizon/data/alignments/RAxML_info.Makona_600_gp.raxml.tree\n",
      "Best-scoring ML tree written to: /Users/evogytis/Documents/genomic-horizon/data/alignments/RAxML_bestTree.Makona_600_gp.raxml.tree\n",
      "\n",
      "Overall execution time: 95.354338 secs or 0.026487 hours or 0.001104 days\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /Users/evogytis/Documents/genomic-horizon/data/alignments/\n",
    "\n",
    "alns=*.fasta\n",
    "echo $alns\n",
    "\n",
    "for aln in $alns\n",
    "do\n",
    "    echo $aln\n",
    "    \n",
    "    partition=${aln/.fasta/.part} ## partition designations for each alignment\n",
    "    out=${aln/.fasta/.raxml.tree} ## output format\n",
    "    \n",
    "    raxml -f d -T 2 -m GTRCAT -q $partition -c 25 -p 666 -n $out --silent -s $aln ## run raxml\n",
    "    \n",
    "done;\n",
    "\n",
    "trees=RAxML_bestTree*\n",
    "for tree in $trees\n",
    "do\n",
    "    mv $tree /Users/evogytis/Documents/genomic-horizon/data/trees/${tree/RAxML_bestTree./} ## simplify name of best tree\n",
    "done;"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove RAxML polytomies."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-18T19:34:31.267972Z",
     "start_time": "2018-06-18T19:34:23.749086Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Tree height: 0.001773\n",
      "Tree length: 0.068033\n",
      "\n",
      "Numbers of objects in tree: 809 (209 nodes and 600 leaves)\n",
      "\n",
      "\n",
      "Tree height: 0.003497\n",
      "Tree length: 0.079726\n",
      "\n",
      "Numbers of objects in tree: 641 (41 nodes and 600 leaves)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for gene in ['genome','gp']:\n",
    "    tree_path='/Users/evogytis/Documents/genomic-horizon/data/trees/Makona_600_%s.raxml.tree'%(gene)\n",
    "    \n",
    "    ll=bt.tree()\n",
    "    bt.make_tree(open(tree_path,'r').readline(),ll)\n",
    "    ll.traverse_tree()\n",
    "    ll=ll.collapseBranches(lambda x:x.length<=1e-5)\n",
    "    ll.treeStats()\n",
    "    \n",
    "    out_tree=open(tree_path,'w')\n",
    "    out_tree.write('%s\\n'%(ll.toString(numName=True)))\n",
    "    out_tree.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Infer time trees with TreeTime, save time trees and reconstructed ancestral sequences."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-18T20:01:35.372449Z",
     "start_time": "2018-06-18T19:34:35.661491Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "0.00\t-TreeAnc: set-up\n",
      "\n",
      "5.71\t-TreeAnc: making reduced alignment...\n",
      "\n",
      "7.34\t-TreeAnc: finished reduced alignment...\n",
      "7.37\t--TreeTime.reroot_to_best_root: searching for the best root position...\n",
      "7.52\t--TreeTime.reroot_to_best_root: branch length of children of new root: 4.503e-05, 6.097e-05\n",
      "\n",
      "7.52\t-TreeTime.reroot: with method or node: Clade\n",
      "7.53\t--TreeTime.reroot: Tree is being re-rooted to node new_node\n",
      "\n",
      "7.61\t-TreeAnc.optimize_sequences_and_branch_length: sequences...\n",
      "\n",
      "7.61\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "7.62\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "11.41\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "\n",
      "11.77\t-TreeAnc.optimize_branch_length: running branch length optimization...\n",
      "\n",
      "14.19\t-TreeAnc.prune_short_branches: pruning short branches (max prob at zero)...\n",
      "\n",
      "14.31\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "14.32\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "17.51\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "17.97\t--TreeAnc.optimize_sequences_and_branch_length: Iteration 1. #Nuc changed since prev reconstructions: 0\n",
      "17.97\t--TreeAnc.optimize_sequences_and_branch_length: Unconstrained sequence LH:-41827.526329\n",
      "\n",
      "17.98\t-TreeTime.reroot: with method or node: best\n",
      "17.99\t--TreeTime.reroot_to_best_root: searching for the best root position...\n",
      "18.08\t--TreeTime.reroot_to_best_root: branch length of children of new root: 9.444e-05, 1.090e-05\n",
      "18.08\t--TreeTime.reroot: Tree is being re-rooted to node new_node\n",
      "\n",
      "18.15\t-TreeAnc.optimize_sequences_and_branch_length: sequences...\n",
      "\n",
      "18.15\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "18.16\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "22.34\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "\n",
      "23.07\t-TreeAnc.optimize_branch_length: running branch length optimization...\n",
      "\n",
      "25.91\t-TreeAnc.prune_short_branches: pruning short branches (max prob at zero)...\n",
      "\n",
      "26.02\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "26.03\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "29.41\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "29.76\t--TreeAnc.optimize_sequences_and_branch_length: Iteration 1. #Nuc changed since prev reconstructions: 0\n",
      "29.77\t--TreeAnc.optimize_sequences_and_branch_length: Unconstrained sequence LH:-41827.526329\n",
      "\n",
      "29.77\t###TreeTime.run: INITIAL ROUND\n",
      "\n",
      "29.77\t-ClockTree: Maximum likelihood tree optimization with temporal constraints:\n",
      "29.77\t--ClockTree.init_date_constraints...\n",
      "41.39\t--ClockTime.date2dist: Setting new molecular clock. rate=8.034e-04, R^2=0.6864\n",
      "41.43\t--ClockTree - Joint reconstruction:  Propagating leaves -> root...\n",
      "82.03\t--ClockTree - Joint reconstruction:  Propagating root -> leaves...\n",
      "\n",
      "82.14\t###TreeTime.run: ITERATION 1 out of 5 iterations\n",
      "\n",
      "82.14\t-TreeTime.run: adding coalescent prior with Tc=skyline\n",
      "85.49\t--optimized Tc to 0.008904\n",
      "\n",
      "85.85\t-ClockTree: Maximum likelihood tree optimization with temporal constraints:\n",
      "85.85\t--ClockTree.init_date_constraints...\n",
      "96.86\t--ClockTime.date2dist: Setting new molecular clock. rate=8.034e-04, R^2=0.6864\n",
      "96.90\t--ClockTree - Joint reconstruction:  Propagating leaves -> root...\n",
      "183.22\t--ClockTree - Joint reconstruction:  Propagating root -> leaves...\n",
      "\n",
      "183.45\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "183.46\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "187.30\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "\n",
      "187.78\t###TreeTime.run: ITERATION 2 out of 5 iterations\n",
      "\n",
      "187.78\t-TreeTime.run: adding coalescent prior with Tc=skyline\n",
      "191.10\t--optimized Tc to 0.006979\n",
      "\n",
      "191.48\t-ClockTree: Maximum likelihood tree optimization with temporal constraints:\n",
      "191.48\t--ClockTree.init_date_constraints...\n",
      "202.37\t--ClockTime.date2dist: Setting new molecular clock. rate=8.034e-04, R^2=0.6864\n",
      "202.41\t--ClockTree - Joint reconstruction:  Propagating leaves -> root...\n",
      "292.68\t--ClockTree - Joint reconstruction:  Propagating root -> leaves...\n",
      "\n",
      "292.94\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "292.94\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "296.56\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "\n",
      "297.06\t###TreeTime.run: ITERATION 3 out of 5 iterations\n",
      "\n",
      "297.06\t-TreeTime.run: adding coalescent prior with Tc=skyline\n",
      "302.27\t--optimized Tc to 0.006523\n",
      "\n",
      "302.72\t-ClockTree: Maximum likelihood tree optimization with temporal constraints:\n",
      "302.72\t--ClockTree.init_date_constraints...\n",
      "314.23\t--ClockTime.date2dist: Setting new molecular clock. rate=8.034e-04, R^2=0.6864\n",
      "314.28\t--ClockTree - Joint reconstruction:  Propagating leaves -> root...\n",
      "400.46\t--ClockTree - Joint reconstruction:  Propagating root -> leaves...\n",
      "\n",
      "400.69\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "400.69\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "404.12\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "\n",
      "404.57\t###TreeTime.run: ITERATION 4 out of 5 iterations\n",
      "\n",
      "404.57\t-TreeTime.run: adding coalescent prior with Tc=skyline\n",
      "407.54\t--optimized Tc to 0.006621\n",
      "\n",
      "407.89\t-ClockTree: Maximum likelihood tree optimization with temporal constraints:\n",
      "407.89\t--ClockTree.init_date_constraints...\n",
      "418.05\t--ClockTime.date2dist: Setting new molecular clock. rate=8.034e-04, R^2=0.6864\n",
      "418.14\t--ClockTree - Joint reconstruction:  Propagating leaves -> root...\n",
      "504.09\t--ClockTree - Joint reconstruction:  Propagating root -> leaves...\n",
      "\n",
      "504.32\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "504.32\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "507.76\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "\n",
      "508.21\t###TreeTime.run: ITERATION 5 out of 5 iterations\n",
      "\n",
      "508.21\t-TreeTime.run: adding coalescent prior with Tc=skyline\n",
      "508.34\t--Coalescent:optimize_skyline:... current LH: 2345.250310\n",
      "588.52\t--Coalescent:optimize_skyline:...done. new LH: 4968.919815\n",
      "588.52\t--optimized a skyline \n",
      "\n",
      "589.04\t-ClockTree: Maximum likelihood tree optimization with temporal constraints:\n",
      "589.04\t--ClockTree.init_date_constraints...\n",
      "600.41\t--ClockTime.date2dist: Setting new molecular clock. rate=8.034e-04, R^2=0.6864\n",
      "600.50\t--ClockTree - Joint reconstruction:  Propagating leaves -> root...\n",
      "678.49\t--ClockTree - Joint reconstruction:  Propagating root -> leaves...\n",
      "\n",
      "678.71\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "678.72\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "681.85\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "\n",
      "682.36\t###TreeTime.run: FINAL ROUND - confidence estimation via marginal reconstruction\n",
      "\n",
      "682.36\t-ClockTree: Maximum likelihood tree optimization with temporal constraints:\n",
      "682.36\t--ClockTree.init_date_constraints...\n",
      "692.43\t--ClockTime.date2dist: Setting new molecular clock. rate=8.034e-04, R^2=0.6864\n",
      "692.47\t--ClockTree - Marginal reconstruction:  Propagating leaves -> root...\n",
      "787.06\t--ClockTree - Marginal reconstruction:  Propagating root -> leaves...\n",
      "1064.31\t--ClockTree.branch_length_to_years: setting node positions in units of years\n",
      "\n",
      "0.00\t-TreeAnc: set-up\n",
      "\n",
      "0.68\t-TreeAnc: making reduced alignment...\n",
      "\n",
      "0.86\t-TreeAnc: finished reduced alignment...\n",
      "0.88\t--TreeTime.reroot_to_best_root: searching for the best root position...\n",
      "0.98\t--TreeTime.reroot_to_best_root: branch length of children of new root: 1.698e-03, 7.814e-04\n",
      "\n",
      "0.98\t-TreeTime.reroot: with method or node: Clade\n",
      "0.98\t--TreeTime.reroot: Tree is being re-rooted to node new_node\n",
      "\n",
      "1.02\t-TreeAnc.optimize_sequences_and_branch_length: sequences...\n",
      "\n",
      "1.02\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "1.03\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "1.46\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "\n",
      "1.56\t-TreeAnc.optimize_branch_length: running branch length optimization...\n",
      "\n",
      "5.00\t-TreeAnc.prune_short_branches: pruning short branches (max prob at zero)...\n",
      "\n",
      "5.02\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "5.03\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "5.47\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "5.59\t--TreeAnc.optimize_sequences_and_branch_length: Iteration 1. #Nuc changed since prev reconstructions: 0\n",
      "5.60\t--TreeAnc.optimize_sequences_and_branch_length: Unconstrained sequence LH:-4133.480613\n",
      "\n",
      "5.60\t-TreeTime.reroot: with method or node: best\n",
      "5.61\t--TreeTime.reroot_to_best_root: searching for the best root position...\n",
      "5.87\t--TreeTime.reroot_to_best_root: branch length of children of new root: 1.683e-03, 8.108e-04\n",
      "5.87\t--TreeTime.reroot: Tree is being re-rooted to node new_node\n",
      "\n",
      "5.94\t-TreeAnc.optimize_sequences_and_branch_length: sequences...\n",
      "\n",
      "5.94\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "5.95\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6.87\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "\n",
      "7.20\t-TreeAnc.optimize_branch_length: running branch length optimization...\n",
      "\n",
      "9.45\t-TreeAnc.prune_short_branches: pruning short branches (max prob at zero)...\n",
      "\n",
      "9.46\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "9.47\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "9.87\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "9.97\t--TreeAnc.optimize_sequences_and_branch_length: Iteration 1. #Nuc changed since prev reconstructions: 0\n",
      "9.98\t--TreeAnc.optimize_sequences_and_branch_length: Unconstrained sequence LH:-4137.403371\n",
      "\n",
      "9.98\t###TreeTime.run: INITIAL ROUND\n",
      "\n",
      "9.98\t-ClockTree: Maximum likelihood tree optimization with temporal constraints:\n",
      "9.98\t--ClockTree.init_date_constraints...\n",
      "16.73\t--ClockTime.date2dist: Setting new molecular clock. rate=7.441e-04, R^2=0.1387\n",
      "16.76\t--ClockTree - Joint reconstruction:  Propagating leaves -> root...\n",
      "27.57\t--ClockTree - Joint reconstruction:  Propagating root -> leaves...\n",
      "\n",
      "27.66\t###TreeTime.run: ITERATION 1 out of 5 iterations\n",
      "\n",
      "27.66\t-TreeTime.run: adding coalescent prior with Tc=skyline\n",
      "29.60\t--optimized Tc to 0.061018\n",
      "\n",
      "29.88\t-ClockTree: Maximum likelihood tree optimization with temporal constraints:\n",
      "29.88\t--ClockTree.init_date_constraints...\n",
      "36.78\t--ClockTime.date2dist: Setting new molecular clock. rate=7.441e-04, R^2=0.1387\n",
      "36.82\t--ClockTree - Joint reconstruction:  Propagating leaves -> root...\n",
      "54.57\t--ClockTree - Joint reconstruction:  Propagating root -> leaves...\n",
      "\n",
      "54.73\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "54.74\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "55.16\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "\n",
      "55.34\t###TreeTime.run: ITERATION 2 out of 5 iterations\n",
      "\n",
      "55.34\t-TreeTime.run: adding coalescent prior with Tc=skyline\n",
      "57.22\t--optimized Tc to 0.066436\n",
      "\n",
      "57.48\t-ClockTree: Maximum likelihood tree optimization with temporal constraints:\n",
      "57.48\t--ClockTree.init_date_constraints...\n",
      "64.41\t--ClockTime.date2dist: Setting new molecular clock. rate=7.441e-04, R^2=0.1387\n",
      "64.44\t--ClockTree - Joint reconstruction:  Propagating leaves -> root...\n",
      "79.98\t--ClockTree - Joint reconstruction:  Propagating root -> leaves...\n",
      "\n",
      "80.18\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "80.19\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "80.72\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "\n",
      "80.91\t###TreeTime.run: ITERATION 3 out of 5 iterations\n",
      "\n",
      "80.91\t-TreeTime.run: adding coalescent prior with Tc=skyline\n",
      "82.81\t--optimized Tc to 0.067712\n",
      "\n",
      "83.06\t-ClockTree: Maximum likelihood tree optimization with temporal constraints:\n",
      "83.06\t--ClockTree.init_date_constraints...\n",
      "89.72\t--ClockTime.date2dist: Setting new molecular clock. rate=7.441e-04, R^2=0.1387\n",
      "89.76\t--ClockTree - Joint reconstruction:  Propagating leaves -> root...\n",
      "106.92\t--ClockTree - Joint reconstruction:  Propagating root -> leaves...\n",
      "\n",
      "107.16\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "107.17\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "107.59\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "\n",
      "107.77\t###TreeTime.run: ITERATION 4 out of 5 iterations\n",
      "\n",
      "107.77\t-TreeTime.run: adding coalescent prior with Tc=skyline\n",
      "109.65\t--optimized Tc to 0.070116\n",
      "\n",
      "109.91\t-ClockTree: Maximum likelihood tree optimization with temporal constraints:\n",
      "109.91\t--ClockTree.init_date_constraints...\n",
      "116.32\t--ClockTime.date2dist: Setting new molecular clock. rate=7.441e-04, R^2=0.1387\n",
      "116.42\t--ClockTree - Joint reconstruction:  Propagating leaves -> root...\n",
      "131.17\t--ClockTree - Joint reconstruction:  Propagating root -> leaves...\n",
      "\n",
      "131.32\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "131.33\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "131.74\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "\n",
      "131.92\t###TreeTime.run: ITERATION 5 out of 5 iterations\n",
      "\n",
      "131.92\t-TreeTime.run: adding coalescent prior with Tc=skyline\n",
      "132.02\t--Coalescent:optimize_skyline:... current LH: -18939.281792\n",
      "185.89\t--Coalescent:optimize_skyline:...done. new LH: 4220.659051\n",
      "185.89\t--optimized a skyline \n",
      "\n",
      "186.15\t-ClockTree: Maximum likelihood tree optimization with temporal constraints:\n",
      "186.15\t--ClockTree.init_date_constraints...\n",
      "192.53\t--ClockTime.date2dist: Setting new molecular clock. rate=7.441e-04, R^2=0.1387\n",
      "192.57\t--ClockTree - Joint reconstruction:  Propagating leaves -> root...\n",
      "208.96\t--ClockTree - Joint reconstruction:  Propagating root -> leaves...\n",
      "\n",
      "209.12\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "209.12\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "209.53\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "\n",
      "209.71\t###TreeTime.run: FINAL ROUND - confidence estimation via marginal reconstruction\n",
      "\n",
      "209.71\t-ClockTree: Maximum likelihood tree optimization with temporal constraints:\n",
      "209.71\t--ClockTree.init_date_constraints...\n",
      "216.46\t--ClockTime.date2dist: Setting new molecular clock. rate=7.441e-04, R^2=0.1387\n",
      "216.50\t--ClockTree - Marginal reconstruction:  Propagating leaves -> root...\n",
      "239.69\t--ClockTree - Marginal reconstruction:  Propagating root -> leaves...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/lib/python2.7/site-packages/treetime/distribution.py:258: RuntimeWarning: overflow encountered in exp\n",
      "  return np.exp(-1 * self.__call__(x))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "555.29\t--ClockTree.branch_length_to_years: setting node positions in units of years\n"
     ]
    }
   ],
   "source": [
    "root_path='/Users/evogytis/Documents/genomic-horizon/data/'\n",
    "trees={}\n",
    "tip_dates={}\n",
    "for gene in ['genome','gp']:\n",
    "    tree_path='/Users/evogytis/Documents/genomic-horizon/data/trees/Makona_600_%s.raxml.tree'%(gene)\n",
    "    \n",
    "    ll=bt.tree()\n",
    "    bt.make_tree(open(tree_path,'r').readline(),ll)\n",
    "    tip_dates[gene]={k.numName:bt.decimalDate(k.numName.split('|')[-1]) if k.numName not in masked_set else (2013.91506849,2015.91506849) for k in ll.Objects if k.branchType=='leaf'}\n",
    "    \n",
    "    tt = treetime.TreeTime(dates=tip_dates[gene], tree=root_path+'trees/Makona_600_%s.raxml.tree'%(gene), aln=root_path+'alignments/Makona_600_%s.fasta'%(gene), gtr='HKY85')\n",
    "    trees[gene]=tt\n",
    "    \n",
    "    new_root=tt.reroot_to_best_root()\n",
    "    new_root.mutation_length=0.0\n",
    "    tt.reroot(new_root)\n",
    "    treetime.treetime.treetime_to_newick(tt,root_path+'trees/Makona_600_%s.rooted.tree'%(gene)) ## save tree\n",
    "    \n",
    "    tt.run(root='best', infer_gtr=False, Tc='skyline', resolve_polytomies=False, max_iter=5,debug=True,time_marginal=True)\n",
    "    \n",
    "    tt.branch_length_to_years()\n",
    "    \n",
    "    treetime.treetime.treetime_to_newick(tt,root_path+'trees/Makona_600_%s.tt_dated.tree'%(gene)) ## save tree"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Infer mutations on a genome phylogeny."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-18T21:16:00.864119Z",
     "start_time": "2018-06-18T21:15:41.930231Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "0.00\t-TreeAnc: set-up\n",
      "\n",
      "5.84\t-TreeAnc: making reduced alignment...\n",
      "\n",
      "7.49\t-TreeAnc: finished reduced alignment...\n",
      "\n",
      "7.52\t-TreeAnc.infer_ancestral_sequences: method: ml\n",
      "\n",
      "7.52\t-TreeAnc inferring the GTR model from the tree...\n",
      "7.52\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "11.12\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "11.48\t--TreeAnc.infer_gtr: counting mutations...\n",
      "\n",
      "14.17\t-GTR: with alphabet: ['A' 'C' 'G' 'T']\n",
      "\n",
      "14.17\t-GTR: model inference \n",
      "14.18\t--TreeAnc.infer_gtr: setting overall rate to 1.0...\n",
      "14.19\t--TreeAnc._ml_anc_joint: type of reconstruction: Joint\n",
      "17.43\t--TreeAnc._store_compressed_sequence_pairs...\n",
      "17.83\t--TreeAnc.get_reconstructed_alignment ...\n"
     ]
    }
   ],
   "source": [
    "root_path='/Users/evogytis/Documents/genomic-horizon/data/'\n",
    "\n",
    "tree_path=root_path+'trees/Makona_600_genome.rooted.tree'\n",
    "\n",
    "ll=bt.tree()\n",
    "bt.make_tree(open(tree_path,'r').readline(),ll)\n",
    "\n",
    "tip_dates={k.numName:bt.decimalDate(k.numName.split('|')[-1]) if k.numName not in masked_set else (2013.91506849,2015.91506849) for k in ll.Objects if k.branchType=='leaf'}\n",
    "    \n",
    "tt = treetime.TreeTime(dates=tip_dates, tree=tree_path, aln=root_path+'alignments/Makona_600_genome.fasta', gtr='HKY85')\n",
    "# tt.run(root='best', infer_gtr=True, Tc='skyline', resolve_polytomies=False, max_iter=10,debug=True,time_marginal=True)\n",
    "tt.infer_ancestral_sequences(method='ml',infer_gtr=True)\n",
    "\n",
    "aln=tt.get_reconstructed_alignment()\n",
    "aln_out=open('/Users/evogytis/Documents/genomic-horizon/data/alignments/Makona_600_genome.anc.fasta','w')\n",
    "for seq in aln:\n",
    "    aln_out.write('>%s\\n%s\\n'%(seq.id,seq.seq))\n",
    "aln_out.close()\n",
    "treetime.treetime.treetime_to_newick(tt,root_path+'trees/Makona_600_genome.anc.tree') ## clades receive different labels after ancestral state reconstruction, so we need to save it again, even though its topology is identical to the starting tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-18T21:18:49.686467Z",
     "start_time": "2018-06-18T21:18:49.659915Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "600\n"
     ]
    }
   ],
   "source": [
    "out_locs=open('/Users/evogytis/Documents/genomic-horizon/data/Makona_600_locs.masked.txt','w')\n",
    "out_locs.write('strain,loc\\n')\n",
    "accession_to_strain={}\n",
    "\n",
    "for line in open('/Users/evogytis/Documents/genomic-horizon/data/alignments/Makona_600_gp.dates.txt','r'):\n",
    "    strain,date=line.strip('\\n').split(',')\n",
    "    loc=strain.split('|')[4]\n",
    "    accession='|'.join(strain.split('|')[1:3])\n",
    "    accession_to_strain[accession]=strain\n",
    "    if loc=='WesternArea':\n",
    "        loc='WesternUrban'\n",
    "    elif loc=='':\n",
    "        loc='?'\n",
    "        \n",
    "    if strain in masked_set:\n",
    "        out_locs.write('%s,?\\n'%(strain))\n",
    "    else:\n",
    "        out_locs.write('%s,%s\\n'%(strain,loc))\n",
    "    \n",
    "print len(accession_to_strain)\n",
    "out_locs.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-15T22:00:52.191346Z",
     "start_time": "2019-03-15T22:00:47.622770Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Makona_600_genome.tt_dated.tree\n",
      "\n",
      "Completed mugration model inference of attribute 'loc' for Makona_600_genome.tt_dated.tree\n",
      "\n",
      "Saved inferred mugration model as: 2019-03-15_mugration/GTR.txt\n",
      "Saved table with ancestral state confidences as: 2019-03-15_mugration/confidence.csv\n",
      "Saved annotated tree as: 2019-03-15_mugration/annotated_tree.nexus\n",
      "Makona_600_gp.tt_dated.tree\n",
      "\n",
      "Completed mugration model inference of attribute 'loc' for Makona_600_gp.tt_dated.tree\n",
      "\n",
      "Saved inferred mugration model as: 2019-03-15_mugration/GTR.txt\n",
      "Saved table with ancestral state confidences as: 2019-03-15_mugration/confidence.csv\n",
      "Saved annotated tree as: 2019-03-15_mugration/annotated_tree.nexus\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda2/lib/python2.7/site-packages/treetime/treeanc.py:1425: RuntimeWarning: divide by zero encountered in log\n",
      "  node.marginal_outgroup_LH, pre = normalize_profile(np.log(node.up.marginal_profile) - node.marginal_log_Lx,\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /Users/evogytis/Documents/genomic-horizon/data/trees\n",
    "trees=$(ls -d *dated.tree)\n",
    "\n",
    "locs=/Users/evogytis/Documents/genomic-horizon/data/Makona_600_locs.masked.txt\n",
    "\n",
    "for tree in $trees\n",
    "    do echo $tree\n",
    "    /Users/evogytis/Documents/treetime/bin/treetime mugration --tree $tree --states $locs --confidence --attribute \"loc\"\n",
    "    mv 2019-03-15_mugration/annotated_tree.nexus ${tree/.tt_dated/.tt_dated_traited}\n",
    "    mv 2019-03-15_mugration/confidence.csv ${tree/.tt_dated.tree/.trait_confidence.txt}\n",
    "done;"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-18T21:13:22.519647Z",
     "start_time": "2018-04-18T21:13:20.397031Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Tree height: 0.002770\n",
      "Tree length: 0.068720\n",
      "multitype tree\n",
      "annotations present\n",
      "\n",
      "Numbers of objects in tree: 810 (210 nodes and 600 leaves)\n",
      "\n",
      "\n",
      "Tree height: 0.004500\n",
      "Tree length: 0.080780\n",
      "multitype tree\n",
      "annotations present\n",
      "\n",
      "Numbers of objects in tree: 642 (42 nodes and 600 leaves)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for gene in ['genome','gp']:\n",
    "    tree_path='/Users/evogytis/Documents/genomic-horizon/data/trees/Makona_600_%s.rooted.tree'%(gene)\n",
    "    \n",
    "    ll=bt.tree()\n",
    "    bt.make_tree(open(tree_path,'r').readline(),ll)\n",
    "    ll.treeStats()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
